src/palace.pyx

```diff
@@ -121,10 +121,17 @@ def query_extension(name: str) -> bool:
     return devmgr.query_extension(name)
 
 
-def current_context() -> Optional[Context]:
-    """Return the context that is currently used."""
+def current_context(thread: bool = False) -> Optional[Context]:
+    """Return the context that is currently used.
+
+    If `thread` is set to `True`,
+    it will return the current context thread.
```

Comment:
Something is wrong with the phrasing here. Please see alure's doc and also document the exception might be raised.

Reply:
Is the phrasing ok now?

Reply:
Readable to me, however *Default value for `thread` is `False`* seems a bit redundant.  Mind the spacing between sentences (of all doctrings) though and please also add the caution you had in `use_context` (I believe it's for both, and palace doesn't have DeviceManager).

Reply:
So I'll just remove DeviceManager out of the doc? Btw should I put *Require* instead of *Requires* here?

Reply:
I think you can make reference to the function**s** directly; PEP 257 recommends using infinitive forms.

Reply:
Which functions are referred to here?

Reply:
`query_extension`.  Thank you for asking tho, on second thought you can mention the device manager thing as *module level*.